paper_framework_sources/p2203_bulk_schedule.md

```diff
@@ -0,0 +1,416 @@
+
+All of the `many_receiver`'s `set_next` calls shall happen-before any of the `many_receiver`'s completion-signal operations.
+Exactly one of the receiver’s completion-signal operations shall complete non-exceptionally before the receiver is destroyed.
+
+If any call to `execution::set_next` or `execution::set_value` exits with an exception, it is still valid to call `execution::set_error` or `execution::set_done` on the `many_receiver`.
+If all calls to `execution::set_next` complete successfully, it is valid to call `execution::set_value` on the `many_receiver`.
+
+Once one of a `many_receiver`’s completion-signal operations has completed non-exceptionally, the `many_receiver` contract has been satisfied.
+
+
+## Concepts many_sender and many_sender_to
+XXX TODO The many_sender and many_sender_to concepts…
+```
+    template<class S>
+      concept many_sender =
+        move_constructible<remove_cvref_t<S>> &&
+        !requires {
+          typename sender_traits<remove_cvref_t<S>>::__unspecialized; // exposition only
+        };
+
+    template<class S, class R>
+      concept many_sender_to =
+        many_sender<S> &&
+        many_receiver<R> &&
+        requires (S&& s, R&& r) {
+          execution::connect((S&&) s, (R&&) r);
+        };
+```
+
+None of these operations shall introduce data races as a result of concurrent invocations of those functions from different threads.
+
+A `many_sender` type’s destructor shall not block pending completion of the submitted function objects. [Note: The ability to wait for completion of submitted function objects may be provided by the associated execution context. –end note]
+
+
+# Why use get_(bulk_)execution_policy on the receiver?
+There are two questions embedded in this:
+
+ * Why pass the policy into the bulk operation at all (with reference to the discussion point in [@P2181])?
+ * Why do it this way rather than as a parameter?
+
+
+## Why pass a policy into the bulk API?
+For a compound algorithm, it is unreasonable to assume that the policy passed to the algorithm need be the one applied to the executor.
+This is for a variety of reasons, but primarily that the author of the algorithm may not be in a position to match policies perfectly.
+We might implement sort:
+```
+output_sender std::execution::sort(input_sender, executor, policy, comparison_function, range);
+```
+
+where the policy passed to match comparison_function is `par_unseq` but where a complex multi-stage sort needs par internally.
+Here we end up with a question: do we have to restrict the interface to match the implementation? That might be hard.
+The alternative is to transform the executor inside the algorithm.
+Potentially using `require`, or a `with_policy` wrapper.
+The problem is that this is semantically identical to passing the policy with the continuation, but syntactically worse:
```

Comment:
I am having trouble following this paragraph. Can you reword or add some extra explainations here?